### DISCUSSION TITLE
Some more Record characters

### DISCUSSION THREAD

**Discussion by TheKnightWho** [psb1558/Junicode-font#112] [Ideas]
Hi - I was wondering if you could please incorporate some further Record type characters? I am currently doing a transcription of Statutes of the Realm (among others), and have run into a small number of issues.

Rather than giving you a big list, it's probably easier for me to link to three further contemporary printer's lists.

1. [The History and Antiquities of the County of Dorset](https://archive.org/details/b30456496_0001/page/n123/mode/2up) (1774). This is the first book published using Record type, and contains an extract from the Domesday book by Abraham Farley as well as a guide to the abbreviations. This was used as a testbed for Farley's 1783 edition of Domesday. This is considerably more complex than conventional Record type, with several unique glyphs (some of which aren't listed there, but laid out in the works below). Sadly, the matrices were destroyed in a fire in 1808.

2. [Typographia](https://archive.org/details/typographiaorpri02john/page/248/mode/2up) (1824) Vol. 2 pp. 248 to 259. This has quite a few that aren't listed anywhere else, including a double l with combining double tilde overlay, which is in Statutes of the Realm Appendix D (but looks at first glance like the Middle Welsh ỻ).

3. [A Dictionary of the Art of Printing](https://archive.org/details/cu31924029491606/page/676/mode/2up) (1841), pp. 677 to 679. This has a smaller list, but some (such as the e with flourish for 'est' and the m for 'manucaptor') being new.

**Comment by psb1558** [comment]
At a glance, it looks like most of this is already in Junicode, so it would be a matter of adding to the guide I've made on this subject. But I'll go through these pages carefully, looking for new stuff.

**Comment by TheKnightWho** [comment]
Thanks. I've just spotted that this:

> A first go at the common mark of suspension (most often for a missing n or m, and represented by a combining macron: <img alt="image" width="744" src="https://user-images.githubusercontent.com/16215767/137214625-e126e78d-15ee-43c7-9546-fd2e7eaffcab.png"> I don't know how much I like this. I've seen this shape of the mark before, though: I may take a prowl through some early printed books to get a good sample of them.

also seems to crop up in Domesday as an overlay stroke:

![image](https://user-images.githubusercontent.com/6895210/156442544-50bb5cd5-7987-4d8e-9984-485f19b5815b.png) 
![image](https://user-images.githubusercontent.com/6895210/156442602-09eb96b5-5294-44aa-bea7-afd58365fbdd.png)

This is as distinct from the tilde:

![image](https://user-images.githubusercontent.com/6895210/156442690-1950a00b-f91e-4127-bab0-6c0a196b7e64.png)

**Comment by psb1558** [comment]
That's very nice printing in the 1774 Dorset book. The overlay stroke on the b and h is generally handled as an overlay tilde (U+0334), though its shape is a little different. I've see the style of stroke on the uncial M in pre-conquest MSS, but I can't think of anything like it in MUFI or Unicode. It's a genuine variant. I'll work on this and the other things.

**Reply by TheKnightWho** [reply]
Many thanks!

**Reply by TheKnightWho** [reply]
It seems to pop up as a combining stroke, too.

![image](https://user-images.githubusercontent.com/6895210/156446893-068b3348-6725-4584-98fc-77b7f3041777.png)

Seems to resemble [U+1D129](https://www.compart.com/en/unicode/U+1D129), though I'm unsure what the relationship is between them (there no doubt is one).

**Comment by psb1558** [comment]
Here's a [transcript of the list of abbreviations](https://github.com/psb1558/Junicode-font/blob/master/docs/Dorset_1774_abbrevs.pdf) in the Dorset 1774 book. It contains several new things, which will be in the next version of Junicode 2 (v. 1.043):

- cv91 now includes the tilde overlay (U+0334) as well as the short stroke (U+0335): cv91[1] is a wide version and cv91[2] is a shorter version.
- The page has a spacing version of the combining zigzag (U+035B). It's not in Unicode or MUFI, so making it accessible is a challenge. I've made it a variant of the spacing macron (U+00AF) on cv67, which is thus similar to cv84 (variants of the non-spacing macron U+0304).
- The abbreviation in villani (second col. near bottom) is treated as a ligature of l~l~ (the ascii tilde, not the combining mark, becausse it's not practical to make a ligature with combining marks). I'm studying better ways to do this.

I've looked over the other two books, but only in a cursory way. Some of the characters in these lists seem to me improbable, and I'd like to see them actually in use before I include them.

**Reply by TheKnightWho** [reply]
Thanks - that's excellent work, and thank you for the quick turnaround.

I've noticed  that the the l with combining double tilde seems to be set to l U+334 l U+334 (rather than l\~l\~), and it's a default ligature. Is that correct? I don't actually think it's an issue as I can't imagine that combination being used without wanting the double tilde, but I just wanted to check.

With the characters in the other two books, I'll investigate. I have a copy of the full 1783 Domesday which I've been meaning to digitise at some point, so it may be a case that some of these aren't in the Dorset section. I suspect Little Domesday might have some unique stuff, and there's also the 1816 'Additamenta' volume with four further Domesday(-adjacent) manuscripts.

**Comment by psb1558** [comment]
The ligature made with the combining character seemed intuitively right, but I couldn't make it work in hlig. So I put that and the equivalent with bb in ccmp, where forming a ligature with a combining mark is not an issue. You're right that I figured I could get away with it because the combination would be very rare. If anyone needs the two separate combinations, they can always drop in U+200C, zero-width non-joiner.

With Hutchins 1774 I presumed that everything he was reporting in his list of abbrevs was in the Dorset section he prints, and besides, it really looked as if he knew what he was doing. Also, when I checked in Cappelli, I found the same things or at least things that were close enough. So I gave Hutchins the benefit of the doubt.

The lists in the other two come in the context of general works about printing. They sometimes seem to veer towards the naive or mistaken, and I often couldn't verify against Cappelli or other sources. So I didn't trust them, and ended up including only the bb + tilde ligature, which I was able to verify.

**Reply by TheKnightWho** [reply]
Thanks - that makes a lot of sense.

Just to be pedantic, it's Abraham Farley that you want to be thanking for the Domesday section of the Hutchins edition. Hutchins was merely taking advantage of the opportunity to include it ;-)

**Reply by TheKnightWho** [reply]
Just on this point in particular re the double letter ligatures, I have spotted that it also happens with "dd"  in the 1136 charter mentioned on the other thread as part of an abbreviation for St David:
![image](https://user-images.githubusercontent.com/6895210/159003406-8a8ddb3a-a94c-45d8-a4bc-8b5e1ee054c4.png)

Somewhat arbitrarily, it's rendered in the Record edition as two separate glyphs combining macron overlays:
![image](https://user-images.githubusercontent.com/6895210/159003626-08224652-c6fc-47c0-932e-4d5e076deea3.png)

But the line through the ascenders sloping upwards is, in other instances, represented by the combining tilde overlay, so I think it would make sense for any ligature in Junicode to use that (and also so as to match the other two ligatures of the same kind, too).
![image](https://user-images.githubusercontent.com/6895210/159004013-b417d5db-7e73-44ac-86ef-ca9b8320d469.png)
![image](https://user-images.githubusercontent.com/6895210/159004063-4c88c4f0-9623-48f1-bc26-7d07834627ce.png)
![image](https://user-images.githubusercontent.com/6895210/159004397-09d95840-df46-48c5-ac58-ce1a19410afb.png)
![image](https://user-images.githubusercontent.com/6895210/159004445-3a107530-516a-432d-8b9e-6129c25d6989.png)

**Reply by psb1558** [reply]
This will be handled the way the other double letters with tilde overlay are handled:
<img width="123" alt="image" src="https://user-images.githubusercontent.com/16215767/159070699-88b511db-45af-4942-9856-6135b7dad75a.png">
The two ds with stroke are clearly the printer getting desperate. It's interesting how much more constrained you  are setting a text in type than writing it out with a pen.

**Reply by TheKnightWho** [reply]
Thanks! And yes, I agree. It's surprising how much this seems to have happened, too, given that Statutes was very much "no expenses spared".

**Reply by psb1558** [reply]
I'm going to guess that they were planning the publication while the editing was going on, trying to anticipate all the bits of type that would be needed (there would have been lots, because every base + mark combo had to have its own piece of type). No surprise if they missed a few and had to fake them.

It's a very nice piece of printing.

**Reply by TheKnightWho** [reply]
It took them 10 years to publish volume 1, with the other ten volumes only taking 18 years - and with a reduced editorial team at that. Quite honestly I’m not surprised, given what they were dealing with. The Scottish equivalent, _The Acts of the Parliaments of Scotland_, ended up delaying publication of volume 1 until 1844 due to editorial issues with the manuscripts, with volume 2 having come out in 1814.

I’ve been trying to track down if the matrices survived, but with no luck. I wouldn’t be surprised if they’re sitting in Monotype’s basement or something, given all the type foundry mergers in the last 200 years.

**Comment by TheKnightWho** [comment]
There does seem to be some consistency in Record works when it comes to the long s with flourish, as the below from Statutes of the Realm matches with the reference works above:

![image](https://user-images.githubusercontent.com/6895210/158010141-e2c87e96-bf3e-47ae-b0e8-1123a8bb5b14.png)

I'm unsure if you think this is worth including as a variant, as it's somewhat different to the form specified in MUFI.

**Reply by psb1558** [reply]
MUFI isn't strong in Old French. Can you give me a reference for this? Are MSS of these statutes available online? When looking at  letters with flourishes and such, it helps a lot to see what these printed books are interpreting.

**Reply by TheKnightWho** [reply]
That might be quite tricky - it's vol. 1 p. 227 of Statutes, and is cited as being from MS. Cott. Claudius D. II., and also in MS. Harl. 395, 867 and 947, none of which are online.

I've also spotted it in Appendix H II (pp. lxxxviii-ix) in Latin:
![image](https://user-images.githubusercontent.com/6895210/158172991-771dc739-05d2-4d89-b98e-e78f1a11e569.png)

and also Scots:
![image](https://user-images.githubusercontent.com/6895210/158173100-3d4f31cf-310c-42f6-89c1-61371fa5fd78.png)
![image](https://user-images.githubusercontent.com/6895210/158173165-19a1bff2-71ee-4478-8aa8-8800cad5da90.png)

None of these are from manuscripts I have any idea how to locate.

**Comment by TheKnightWho** [comment]
[Domesday Book Illustrated](https://archive.org/details/domesdaybookillu00kelh) (1788) has some of the signs in the reference works, and is notable as it uses the exact same Record type as used in the 1774 Dorset book. Both were printed by John Nichols, who helped develop the type with Farley.

pp.161, 226 & 342: - this could be an eth, but I'm unsure:
![image](https://user-images.githubusercontent.com/6895210/158012575-673d3338-3e4f-45d2-866b-dec3d0c32577.png)
![image](https://user-images.githubusercontent.com/6895210/158011506-d77a188b-e087-4f8f-b983-3bc8f4d1e430.png)
![image](https://user-images.githubusercontent.com/6895210/158012382-e9721474-0ff9-49cb-abc9-7c470f39dcf8.png)

p. 190 - the numeral 6, which is incorrectly said to be the numeral 5 in Typographia:
![image](https://user-images.githubusercontent.com/6895210/158010709-7e23743c-9233-4d65-a6f3-8d14b654dfb2.png)

Both also on pp. 209 & 273:
![image](https://user-images.githubusercontent.com/6895210/158011360-5423733b-fcb3-4d76-b784-f0fa5e4a9e34.png)
![image](https://user-images.githubusercontent.com/6895210/158011758-ecf965f6-df15-464c-a502-1c8464044e21.png)

p. 201 - the dot over tilde(?) for "est:
![image](https://user-images.githubusercontent.com/6895210/158010815-1e132918-f895-4b6b-8cb4-19db14f361e2.png)

p. 206 - a sign I don't recognise:
![image](https://user-images.githubusercontent.com/6895210/158011273-09d9397b-afeb-4945-b9a9-9393bb4d91b2.png)

p. 222 - this mark for "ver", which seems to be the same as the one on p. 256 of Typographia:
![image](https://user-images.githubusercontent.com/6895210/158011470-53808bad-6004-4492-a4d4-874260c7dffb.png)

p. 239 - unsure:
![image](https://user-images.githubusercontent.com/6895210/158011698-2ea4690e-bac9-422f-87f4-194cbe936baa.png)

pp. 303-4, 317 etc. - a variant of the letter et:
![image](https://user-images.githubusercontent.com/6895210/158012088-63fc7288-fc69-47b6-a99b-658028023f14.png)
![image](https://user-images.githubusercontent.com/6895210/158012045-ef34e96e-fc33-4e0f-831e-ba2b23b5456f.png)
![image](https://user-images.githubusercontent.com/6895210/158012147-102a55c6-0009-464f-a847-a26822b93ac2.png)
![image](https://user-images.githubusercontent.com/6895210/158012197-a997cb63-e214-4c11-b4e4-ab987df38d6e.png)
![image](https://user-images.githubusercontent.com/6895210/158011000-efc77b27-035e-4d2e-941b-1cb34ea26ce6.png)

pp. 284 & 319 - capital S with combining middle tilde:
![image](https://user-images.githubusercontent.com/6895210/158011870-20d2db91-4a51-47d9-b14b-97ddb26cdd53.png)
![image](https://user-images.githubusercontent.com/6895210/158010946-42236b77-e030-4a8a-b3d8-deb89c61da50.png)

p. 352 - another one I'm not sure of:
![image](https://user-images.githubusercontent.com/6895210/158012450-7210c73a-9208-42e3-a879-b50483cdd6f2.png)

p. 317 - a bunch of symbols used:
![image](https://user-images.githubusercontent.com/6895210/158012539-bb55b9d6-2610-4d0d-a144-862b448ea564.png)

**Reply by psb1558** [reply]
Several of these are interesting. I'd like to see the "geldaver" abbreviation in the MS. If you can supply a place where it's actually used, I might be able to find my way back to the MS via Open Domesday.

The usual abbreviation for _vel_ "or" is a crossed l, or u/v + crossed l (as on p. 352). The one on p. 206 might be  a desperate attempt to render that in print, or maybe a rendering of some stylized version of it. Again it would be useful to see what the Domesday MSS have.

You can see how different an MS abbreviation can be from the printed interpretation from, e.g., the crossed S abbreviation for Sanctus (compare to pp. 284 & 319):
<img width="234" alt="image" src="https://user-images.githubusercontent.com/16215767/158017974-f8d49e7f-c461-44a3-b83a-c7256d659252.png">
What I try to do is look at various versions of a symbol or abbreviation and try to figure out a sort of "average" form that will be intelligible to everyone.

For some things, bear in mind that I can't get into the business of imitating other typefaces. The "et" abbreviation, for example, is quite intelligible as an "et" abbreviation, only it has an added flourish (probably imitating one of those infamous gothic squiggles). As long as you map it to the right Unicode (U+A76B), you're fine.

**Reply by TheKnightWho** [reply]
Thanks - I'll have a look to see if I can track down places that these occur in the Farley text. I'm familiar with Open Domesday, so will try to present side by side comparisons. It may take a bit of time. 

In terms of imitating other fonts, you raise a good point. With Record stuff, I do wonder to what extent something starts being a character in and of itself, rather than simply a representation of an abbreviation - especially when we know that many of these are themselves amalgamations due to the limitations of the press. It's the question of "when did MacKenȝie actually become MacKenzie?"

With Statutes of the Realm (and the later revised editions which also include Record type), it has the (somewhat dubious) distinction of being [legally definitive](https://www.legislation.gov.uk/ukpga/1978/30/section/19), and Halsbury's notes that it's very possible that because of that legislation it might very well take precedence over any originals. So you end up with the absurd situation of it [being scanned in](https://www.legislation.gov.uk/aep/Hen3cc1415/52/1) to the modern legislation.gov.uk site...

**Reply by psb1558** [reply]
> With Record stuff, I do wonder to what extent something starts being a character in and of itself, rather than simply a representation of an abbreviation - especially when we know that many of these are themselves amalgamations due to the limitations of the press. It's the question of "when did MacKenȝie actually become MacKenzie?"

Yes, there's a line somewhere, but it's a fuzzy one. A good example, I think, is the Tironian sign for "et" (U+204A). In Junius's "Saxon Pica," which served as the original model for the insular minuscule stuff in Junicode, the sign looks like this:
<img width="59" alt="image" src="https://user-images.githubusercontent.com/16215767/158215781-b5e60ba9-5f02-49c1-bf42-79fac9bd1911.png">
It's not the most common shape found in MSS—that has a straight bottom stroke, kind of like a seven though often less diagonal, e.g.
<img width="52" alt="image" src="https://user-images.githubusercontent.com/16215767/158217968-92d81f19-e71f-437b-8f4a-f4bb08062df8.png">
—but it's close enough. Everyone recognizes it. Move as far forward as the twelfth century, though, and you find a shape that's still the Tironian sign, but hardly recognizable:
<img width="48" alt="image" src="https://user-images.githubusercontent.com/16215767/158216117-7b98c917-7035-465f-ad61-2c39d8028b98.png">
And it sometimes has a cross stroke, as the older Tironian sign never does:
<img width="53" alt="image" src="https://user-images.githubusercontent.com/16215767/158216523-b6c15304-51f7-49a4-8586-bb891fd3949a.png">
The older shape, if used in a transcription of a Middle English text, would be quite jarring. So there are two variants in Junicode. But the number of possible variants is very large. I think one could make a reasonable case that this one is worth a variant:
<img width="146" alt="image" src="https://user-images.githubusercontent.com/16215767/158217067-a256e2ab-dbe8-47ab-8a9e-d3762d92cdbc.png">
But honestly, that way madness lies, and labor without cease. So I have to be rather conservative about variants—sometimes against my own inclination.

**Comment by afarlie** [comment]
In terms of finding older documents for comparison:-
https://archives.parliament.uk/collections has some earlier Acts although not necessarily all the way back to the Middle Ages.

**Reply by psb1558** [reply]
Thanks, I'll explore this.

**Reply by TheKnightWho** [reply]
I have a volume with a bunch of plates of English court hand, too, which I'll have a look through.

**Reply by psb1558** [reply]
Is it more likely to be found in Anglo-French than in English or Latin (thinking of the strange shape of the word "geldaver"—possibly an English word (geld "tax") with a French termination?

**Reply by TheKnightWho** [reply]
Statutes notes that the Latin manuscripts tend to be more heavily abbreviated than the Anglo-Norman, though the overspill into English tends to be at random. Acts of Parliament were handwritten in "chancery hand" and inrolled right up until 1849, so there's a reasonable chance you'll encounter the occasional abbreviation up until that point. You certainly will in the 18th century and earlier with court hand, where you'll run into "Law French".

**Comment by psb1558** [comment]
Sanctus. cv37[4].
<img width="94" alt="image" src="https://user-images.githubusercontent.com/16215767/159077555-abb7b596-0825-4d5c-a6d4-47b096455775.png">
Or better (with an S for comparison):
<img width="194" alt="image" src="https://user-images.githubusercontent.com/16215767/159079382-9ba7e597-1c4b-4ee6-a939-b3a0adb097b7.png">

**Reply by TheKnightWho** [reply]
Thanks! Looks very good.

**Comment by TheKnightWho** [comment]
Some more double overlay tilde combinations (this time from Cappelli);

Double long s:
![image](https://user-images.githubusercontent.com/6895210/159175788-cbec9bdc-f780-43e4-9dbb-d3ca9f337f7c.png)

Double f:
![image](https://user-images.githubusercontent.com/6895210/159175861-aeb28c09-f696-45b0-b78b-80f19f942cea.png)

This second one crops up as another fudge in Statutes for "Roffensis" (p. 44 of the Charters), but I've not been able to locate a facsimile of a 1300 issue of the Charter of the Forest to check it against, annoyingly:
![image](https://user-images.githubusercontent.com/6895210/159175901-1dfb9f5a-9550-4b6e-ac74-c5ce3a9756a4.png)

Double h:
![image](https://user-images.githubusercontent.com/6895210/159176085-f1276aca-2daa-4351-bc5a-ab7b944af487.png)

hd:
![image](https://user-images.githubusercontent.com/6895210/159176132-ee1f6c01-5373-40a7-ba2d-9fb394a6012a.png)

db:
![image](https://user-images.githubusercontent.com/6895210/159176143-3c36ed34-b570-44cb-bc55-a654b78429f6.png)

bd:
![image](https://user-images.githubusercontent.com/6895210/159176309-59d2508c-db1e-4fed-a31f-43cc08b1182c.png)

**Comment by psb1558** [comment]
Can you verify an actual need for these things? I can't add everything in Cappelli, so won't add things on the authority of Cappelli alone.

**Reply by TheKnightWho** [reply]
I was probably getting a bit overexcited. Of those, the double f ligature does make sense to include, but the others perhaps not at the moment.

**Comment by psb1558** [comment]
I doubt there's any way to make this look good, but, well, there it is. It will work like the others.
<img width="134" alt="image" src="https://user-images.githubusercontent.com/16215767/159369307-8513daf7-cd31-4869-b595-5ba12f06f6d6.png">

**Reply by TheKnightWho** [reply]
Thank you - I think it looks good given the constraints. I don't know about the process to propose characters for MUFI, but it does feel like there's enough of a pattern with the double tilde overlay to justify a combining character.

**Reply by psb1558** [reply]
The current MUFI chair is [Tarrin Wills](https://nors.ku.dk/english/staff/?pure=en/persons/559011). I'm rather hesitant about proposing MUFI characters because they tend to get PUA code points, which are seductive because of their apparent straightforwardness, but interfere greatly with accessibility and searchability. Of course, Junicode will offer the accessible OpenType route in any case.

**Reply by TheKnightWho** [reply]
Thanks. I suppose my logic is that while I could probably make a persuasive case that a combining character makes sense here, I'm not really willing to go through the extended rigmarole of trying to get anything added to Unicode itself; something that I'm almost certainly not knowledgeable enough to do, in any event. At least if it's in MUFI, it feels like there's a greater chance that it might get added to Unicode at some point.

**Reply by psb1558** [reply]
I don't know if the tilde combos are good candidates for Unicode encoding. If they look like ligatures, Unicode won't be interested (there are some ligatures from The Old Days, but it's been many years since any new ones have been added)—and these do look like ligatures. They are also reluctant, these days, to add any new base + mark combinations.

The one symbol I've added recently that I'm pretty sure should be included in Unicode is the Sanctus sign, which is similar in function to characters like U+211F Response and U+2123 Versicle, and certainly meets the requirement that it can't easily be composed out of existing Unicode characters. It might be a good idea to scan the Record/Statute axis of texts for characters of this kind and, when you've got a decent collection, see if you can interest someone who is accustomed to making proposals to Unicode and knows the requirements (I know such a person and can pass you the contact information privately).

I don't see MUFI as a necessary intermediate step here, and my position, anyway, is that if it couldn't have a Unicode code point it shouldn't have a MUFI code point.

**Reply by TheKnightWho** [reply]
Sorry - what I meant was proposing the character "combining double tilde overlay", rather than the separate ligatures. I suppose you could argue that they're all ligatures in the sense that they're two letters with the combining tilde overlay next to each other, but I'm not sure that I'd agree, given that we can see typeset work that doesn't combine them like that (as in Statutes where they had to fudge it with "dd" or "ff"), and you could make the same argument that the other double combining characters don't work like that either.

I agree with you on the Sanctus sign, certainly.

In terms of a knowledgeable Unicode contact, I spoke with Michael Everson about this a couple of years ago, but my knowledge of both Unicode and Record type has considerably improved since then, so it might be worth revisiting once I can put together a more mature preliminary proposal.

**Reply by psb1558** [reply]
Sorry, I didn't read with enough attention. Yes, Michael is the person to approach. Writing a Unicode proposal is a specialized and technical business, and he has the know-how.

**Reply by TheKnightWho** [reply]
Thanks - he certainly does seem to know his stuff.

On a related note (and perhaps this is worthy of its own thread), what do you think of the idea of having Unicode variation sequences for some of the things you've implemented through OpenType features? As an example, I understand that Unicode will never agree to encoding the variants of combining zigzag above separately, which I agree with, but I do think there should be some way of specifying that I want the curved version (or whatever) without having to resort to font-specific features. The same goes for quite a few other stylistic choices.

**Reply by jsbien** [reply]
I've already opened the discussion on variation sequences  #122. Please join.

**Reply by TheKnightWho** [reply]
Fab - thanks.
